k9mail/src/main/java/com/fsck/k9/activity/MessageCompose.java

```diff
@@ -956,6 +959,9 @@ public boolean onOptionsItemSelected(MenuItem item) {
             case R.id.read_receipt:
                 onReadReceipt();
                 break;
+            case android.R.id.home:
+                NavUtils.navigateUpFromSameTask(this);
```

Comment:
This method may only be used if the `MessageList` Activity lives in the same task. But `MessageCompose` can be launched by third-party apps into their own task.